### PR TITLE
Add `clients` configuraton key

### PR DESCRIPTION
**Pull request by mainameiz**
Proposal for adding `clients.default` config key.

`config/chewy.yml`

```
production:
  clients:
    default:
      ... parameters to connect to ES cluster

development:
  ...
```

### CODE REVIEW

**Comment by pyromaniac** [comment]
I actually like the approach, it can be even merged separately, awesome job!
As for the implementation itself: are you sure there will be no keys left on the top level?

File: lib/chewy/config.rb [hunk 1]
```diff
@@ -2,6 +2,15 @@ module Chewy
   class Config
     include Singleton
 
+    GLOBAL_OPTIONS = %i[
+      indices_path
+      prefix
+      index
+      wait_for_status
```
Comment (mainameiz) [inline lib/chewy/config.rb]: I suppose this option should be client-specific, but let's keep this option global for now for simplicity

File: lib/chewy/config.rb [hunk 2]
```diff
@@ -144,11 +153,20 @@ def search_class=(base)
     #            number_of_replicas: 0
     #
     def configuration
-      yaml_settings.merge(settings.deep_symbolize_keys).tap do |configuration|
-        configuration[:logger] = transport_logger if transport_logger
-        configuration[:indices_path] ||= indices_path if indices_path
-        configuration.merge!(tracer: transport_tracer) if transport_tracer
+      configuration = yaml_settings.merge(settings.deep_symbolize_keys)
+      unless configuration.key?(:clients)
+        ActiveSupport::Deprecation.warn('Define connection settings under `clients` key. Top level configuration is deprecated.')
```
Comment (mainameiz) [inline lib/chewy/config.rb]: Don't know is it a correct message...

**Comment by mainameiz** [comment]
@pyromaniac Seems like it works :) What do you think about implementation? What should I change?

**Comment by mainameiz** [comment]
There are many deprecation warnings for now, I will fix them after all changes/refactorings in current implementation.

**Comment by mainameiz** [comment]
@pyromaniac Nice to see you again! :) I hope you had a great holiday and hope you will have some time to talk about this feature.

**Comment by pyromaniac** [comment]
Hey! Yeah, sure, I'm almost ready. You know, main focus doesn't let me spend too much time here, unfortunately. What's the next step?

**Comment by jmehnle** [comment]
I'm hoping we can get this unstuck. Anything I can help? @pyromaniac, it seems [@mainameiz wanted to get general feedback](https://github.com/toptal/chewy/pull/613#issuecomment-346861886) from you on his latest version of this PR from Nov 24, 2017.

**Comment by mainameiz** [comment]
https://github.com/toptal/chewy/issues/338

**Comment by jeremyhaile** [comment]
@mainameiz did you ever get this working?

**Pull request closed without merging**